### PR TITLE
Additional Android examples

### PR DESCRIPTION
**Pull request by yanex**

### CODE REVIEW

File: gradle/android-butterknife/app/src/main/java/org/example/kotlin/butterknife/SimpleAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        val word = getItem(position)
+        holder.word.text = "Word: $word"
+        holder.length.text = "Length: ${word.length()}"
+        holder.position.text = "Position: $position"
+
+        return view
+    }
+
+    class ViewHolder(view: View) {
+        @Bind(R.id.word) lateinit val word: TextView
```
Comment (abreslav) [inline gradle/android-butterknife/app/src/main/java/org/example/kotlin/butterknife/SimpleAdapter.kt]: Different style in different places: let's put annotations on the previous line everywhere

File: gradle/android-dagger/app/src/main/java/com/example/dagger/kotlin/DemoApplication.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+    private lateinit var component: ApplicationComponent
+
+    override fun onCreate() {
+        super.onCreate()
+        component = initDaggerComponent()
+        component().inject(this) // As of now, LocationManager should be injected into this.
+    }
+
+    public fun component(): ApplicationComponent = component
```
Comment (abreslav) [inline gradle/android-dagger/app/src/main/java/com/example/dagger/kotlin/DemoApplication.kt]: Why this function? A private setter for `component` would be ok, no?

File: gradle/android-dagger/app/src/main/java/com/example/dagger/kotlin/ForApplication.kt [hunk 3]
```diff
@@ -15,10 +15,8 @@
...
 package com.example.dagger.kotlin
 
-import java.lang.annotation.Retention
-import java.lang.annotation.RetentionPolicy.RUNTIME
 import javax.inject.Qualifier
 
-Qualifier
-Retention(RUNTIME)
+@Qualifier
+@Retention(AnnotationRetention.RUNTIME)
```
Comment (abreslav) [inline gradle/android-dagger/app/src/main/java/com/example/dagger/kotlin/ForApplication.kt]: That's the default now

File: gradle/android-dbflow/app/src/main/java/mobi/porquenao/poc/kotlin/ui/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+public class MainActivity : BaseActivity() {
+
+    lateinit var mAdapter: MainAdapter
+    lateinit var vList: RecyclerView
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.main)
+        vList = findViewById(R.id.list) as RecyclerView
```
Comment (abreslav) [inline gradle/android-dbflow/app/src/main/java/mobi/porquenao/poc/kotlin/ui/MainActivity.kt]: Why not use Android Extensions for this?

File: gradle/android-dbflow/app/src/main/java/mobi/porquenao/poc/kotlin/ui/BaseActivity.kt [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    lateinit var vToolbar: Toolbar
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        prepareTaskDescription()
+    }
+
+    override fun setContentView(layoutResID: Int) {
+        super.setContentView(layoutResID)
+        vToolbar = findViewById(R.id.toolbar) as Toolbar
```
Comment (abreslav) [inline gradle/android-dbflow/app/src/main/java/mobi/porquenao/poc/kotlin/ui/BaseActivity.kt]: Why not use Android Extensions for this?

File: gradle/android-dbflow/app/src/main/java/mobi/porquenao/poc/kotlin/ui/MainAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+    public fun add() {
+        val item = Item()
+        mItems.add(0, item)
+        item.save()
+        notifyItemInserted(0)
+    }
+
+    public class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        val card by lazy { view.findViewById(R.id.card) as CardView }
```
Comment (abreslav) [inline gradle/android-dbflow/app/src/main/java/mobi/porquenao/poc/kotlin/ui/MainAdapter.kt]: It's synchronized now, is it intended?

File: gradle/android-dbflow/app/src/main/java/mobi/porquenao/poc/kotlin/ui/MainActivity.kt (hunk 4 above)
Comment (yanex) [inline gradle/android-dbflow/app/src/main/java/mobi/porquenao/poc/kotlin/ui/MainActivity.kt]: Android Extensions require the corresponding IDEA plugin to be installed.
Also, IMO, sample projects like this should contain as few external dependencies as possible.
Probably it is better to provide a separate example that takes advantage of Android Extensions plugin.

Comment (abreslav) [inline gradle/android-dbflow/app/src/main/java/mobi/porquenao/poc/kotlin/ui/MainActivity.kt]: I'd suggest to add a comment that extension plugin must be installed extensions here. It is important to promote the greatness of our technology, and not send a mixed message: Kotlin is great with Android, but we won't show you that :)

**Pull request merged**